### PR TITLE
fix: Added mock for gitea response in appset PR,SCM generator

### LINKED ISSUE #9287: Use mocks for ApplicationSet Gitea, GitHub, and GitLab unit tests
# Summary

The Gitea unit tests for the PR and SCM generators currently make external API calls. This makes the tests flaky. We should instead use a mocked service.

https://github.com/argoproj/argo-cd/blob/master/applicationset/services/scm_provider/gitea_test.go
https://github.com/argoproj/argo-cd/blob/master/applicationset/services/pull_request/gitea_test.go

# Motivation

Last week integration tests were broken for a couple days because of a change in the tested Gitea org.

# Proposal

We could follow the model of the Bitbucket tests and use a mocked service.

https://github.com/argoproj/argo-cd/blob/1d765a072cfdf3921cf7bf43c353bda61d57347c/applicationset/services/pull_request/bitbucket_server_test.go#L14

### PR DESCRIPTION
**Pull request by rishabh625**
Signed-off-by: rishabh625 <rishabhmishra625@gmail.com>

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)). 

Fixes #9287

### CODE REVIEW

File: applicationset/services/pull_request/gitea_test.go [hunk 1]
```diff
@@ -2,13 +2,38 @@ package pull_request
...
...://gitea.com/api/v1/repos/test-argocd/pr-test/languages","html_url":"https://gitea.com/test-argocd/pr-test","ssh_url":"git@gitea.com:test-argocd/pr-test.git","clone_url":"https://gitea.com/test-argocd/pr-test.git","original_url":"","website":"","stars_count":0,"forks_count":0,"watchers_count":1,"open_issues_count":0,"open_pr_counter":1,"release_counter":0,"default_branch":"main","archived":false,"created_at":"2022-04-06T02:32:09+08:00","updated_at":"2022-04-06T02:33:12+08:00","permissions":{"admin":false,"push":false,"pull":true},"has_issues":true,"internal_tracker":{"enable_time_tracker":true,"allow_only_contributors_to_track_time":true,"enable_issue_dependencies":true},"has_wiki":true,"has_pull_requests":true,"has_projects":true,"ignore_whitespace_conflicts":false,"allow_merge_commits":true,"allow_rebase":true,"allow_rebase_explicit":true,"allow_squash_merge":true,"default_merge_style":"merge","avatar_url":"","internal":false,"mirror_interval":"","mirror_updated":"0001-01-01T00:00:00Z","repo_transfer":null}},"merge_base":"72687815ccba81ef014a96201cc2e846a68789d8","due_date":null,"created_at":"2022-04-06T02:34:24+08:00","updated_at":"2022-04-06T02:34:24+08:00","closed_at":null}]`)
```
Comment (crenshaw-dev) [inline applicationset/services/pull_request/gitea_test.go:25]: Could you run this JSON through a formatter for readability?

File: applicationset/services/scm_provider/gitea_test.go [hunk 2]
```diff
@@ -2,13 +2,125 @@ package scm_provider
...
+					"avatar_url": "",
+					"internal": false,
+					"mirror_interval": "",
+					"mirror_updated": "0001-01-01T00:00:00Z",
+					"repo_transfer": null
+				}]`)
+			if err != nil {
+				t.Fail()
+			}
+		//case "/api/v1/repos/test-argocd/pr-test/labels?limit=0&page=1":
```
Comment (crenshaw-dev) [inline applicationset/services/scm_provider/gitea_test.go:105]: Can this line be dropped?

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]
Looks awesome. Just a couple nitpicks. Thanks @rishabh625!

File: applicationset/services/scm_provider/gitea_test.go (hunk 2 above)
Comment (rishabh625) [reply applicationset/services/scm_provider/gitea_test.go]: Done

**Review by crenshaw-dev — APPROVED** [review]
Thanks @rishabh625 !

**Comment by crenshaw-dev** [comment]
Cherry-picked onto release-2.4 for 2.4.12.

**Pull request merged**